compiler/codegen/symbol.cpp

```diff
@@ -786,6 +786,24 @@ static Type* getArgSymbolCodegenType(ArgSymbol* arg) {
   return useType;
 }
 
+// Alter the C code type to avoid _ref_, c_ptr_ prefixes and instead use a
+// type that C understands without additional information (e.g. instead of
+// _ref_int64_t, just int64_t *).  But only for exported symbols like the
+// return type of exported functions, or arguments of those functions.
+static std::string
+transformTypeForPointer(Type* type) {
+  std::string typeName = type->codegen().c;
+  if (type->symbol->hasFlag(FLAG_REF)) {
+    Type* referenced = type->getValType();
+    return referenced->codegen().c + " *";
+
+  } else if (type->symbol->hasFlag(FLAG_C_PTR_CLASS)) {
+    // Remove the c_ptr_ prefix
+    return typeName.substr(6, std::string::npos) + " *";
```

Comment:
If you look at the implementation for getValType(), for a _ref, it's just getting the type of a field with a particular name. I think you should do the same thing here. Something like `toAggregateType(type)->getField("eltType")->type`.

Reply:
The eltType field no longer exists on the type at codegen because it was a type field and those get removed at resolution.  It looks like it is just included included in the type name of the instantiation

Reply:
Try getDataClassType (I got that by asking "OK, if that's the case, how do we generate the typedefs for a ref?)

Reply:
Refs still have a value field, which isn't just a type.  I'll check the function you recommend, though

Reply:
That worked, good call!